scipy/spatial/transform/_rotation.pyx

```diff
@@ -1865,6 +1873,118 @@ cdef class Rotation:
+        composition : `Rotation` instance
+            If the input Rotation ``p`` contains ``N`` multiple rotations, then
+            the output will contain ``N`` rotations where the ``i``th rotation
+            is equal to ``p[i] ** n``
+
+        Notes
+        -----
+        For example, a power of 2 will double the angle of rotation, and a
+        power of 0.5 will halve the angle. There are two notable cases: if
+        ``n == 1`` then the original rotation is returned, and if ``n == 0``
+        then the identity rotation is returned.
+
+        Note that fractional powers ``0 < n < 1`` which effectively take a root
+        of rotation, do so using the shortest path smallest representation of
+        that angle (the principal root). This means that powers of ``n`` and
+        ``1/n`` are not necessarily inverses of each other. For example, a 0.5
+        power of a +240 degree rotation will be calculated as the 0.5 power of
+        a -120 degree rotation, with the result being a rotation of -60 rather
+        than +120 degrees.
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import Rotation as R
+
+        Raising a rotation to a power:
+
+        >>> p = R.from_rotvec([1, 0, 0])
+        >>> q = p ** 2
+        >>> q.as_rotvec()
+        array([2., 0., 0.])
+        >>> r = p ** 0.5
+        >>> r.as_rotvec()
+        array([0.5, 0., 0.])
+
+        Inverse powers do not necessarily cancel out:
+
+        >>> p = R.from_rotvec([0, 0, 120], degrees=True)
+        >>> ((p ** 2) ** 0.5).as_rotvec(degrees=True)
+        array([  -0.,   -0., -60.])
+        """
+        cdef Py_ssize_t num_rotations = len(self._quat)
+        cdef double angle, scale, angle2, n3, n5
+        cdef double[:, :] result = _empty2(num_rotations, 4)
+        cdef double[:] quat
+
+        if modulus is not None:
+            raise NotImplementedError("modulus not supported")
+
+        if n < 0:
+            raise ValueError("Rotations cannot be raised to negative powers.")
+
+        # short-cuts
+        elif n == 0:
+            return Rotation.identity(num_rotations)
+
+        elif n == 1:
+            result = self._quat.copy()
+
+        elif n == 2:
+            result = _compose_quat(self._quat, self._quat)
+
+        # general scaling of rotation angle
+        else:
+            for ind in range(num_rotations):
+                if self._quat[ind, 3] < 0:  # w > 0 to ensure 0 <= angle <= pi
+                    quat = self._quat[ind, :].copy()
+                    for i in range(4):
+                        quat[i] *= -1
+                else:
+                    quat = self._quat[ind, :]
+                angle = 2 * atan2(_norm3(quat), abs(quat[3]))
+
+                if angle <= 1e-3:  # small angle Taylor series expansion
+                    angle2 = angle * angle
+                    n3 = n * n * n
+                    n5 = n3 * n * n
+                    scale = n + (n - n3) / 24 * angle2 + \
```

Comment:
Another silly question, did you check if there is an actual gain in execution speed? Apart from making the code slightly less readable, I'd imagine this kind of optimization is probably already inside of the cython implementations of `sin`...

Reply:
I didn’t check to see if premultiplying n3 and n5 saved any time, but the Taylor series expansion here is to protect for a divide by 0 error/instability rather than as an optimization.

Reply:
>  the Taylor series expansion here is to protect for a divide by 0 error/instability rather than as an optimization.

Ok I see it now! :)

Reply:
Here's another idea for this: do we know about what value for `norm(quat[:3])` the zero division starts to create numerical problems?

An interesting problem could be to analyze how the error between `sin(n*angle/2) / sin(angle/2)` changes with the number of terms of the Taylor series, according to the distance to `0`, and then use a smaller `eps` so that we can simply state `sin(n*angle/2) / sin(angle/2) = n`